File: Game.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"strings"
	"time"
)

var balance int

func main() {
	fmt.Println("При обнаружении багов пишите мне на почту - pythonerkk@mail.ru")
	fmt.Println("Нужно кинуть скрин недоработки")
	fmt.Println()
	fmt.Println("______________________________________________________")
	fmt.Println()

	rand.Seed(time.Now().UnixNano())
	reader := bufio.NewReader(os.Stdin)

	for {
		playGame(reader)
		fmt.Printf("\nТвой общий счёт = %d\n", balance)
		fmt.Println("\n______________________________________________________\n")
		fmt.Print("Продолжаем игру? (да/нет): ")
		choice, _ := reader.ReadString('\n')
		choice = strings.TrimSpace(strings.ToLower(choice))
		if choice == "нет" {
			break
		}
	}
	fmt.Println("Для закрытия консоли нажми 'ENTER'")
	fmt.Println("Made by MicheZZ")
	reader.ReadString('\n')
}

func playGame(reader *bufio.Reader) {
	fmt.Println("Выбери уровень сложности:")
	fmt.Println("1 уровень от 1 до 100")
	fmt.Println("2 уровень от 1 до 250")
	fmt.Println("3 уровень от 1 до 500")
	fmt.Println("4 уровень от 1 до 1000")
	fmt.Println("5 уровень от 1 до 2000")

	var level int
	for {
		fmt.Print("Введи номер уровня: ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		num, err := strconv.Atoi(input)
		if err == nil && num >= 1 && num <= 5 {
			level = num
			break
		}
		fmt.Println("Ошибка! Введите целое число от 1 до 5")
	}

	var maxNumber, maxAttempts, reward int
	switch level {
	case 1:
		maxNumber = 100
		maxAttempts = 6
		reward = 500
	case 2:
		maxNumber = 250
		maxAttempts = 7
		reward = 800
	case 3:
		maxNumber = 500
		maxAttempts = 8
		reward = 1000
	case 4:
		maxNumber = 1000
		maxAttempts = 9
		reward = 1200
	case 5:
		maxNumber = 2000
		maxAttempts = 10
		reward = 1500
	}

	secret := rand.Intn(maxNumber) + 1
	fmt.Printf("Угадай число от 1 до %d за %d попыток. За лишнюю попытку у тебя будет сниматься по 100 очков\n", maxNumber, maxAttempts)
	fmt.Printf("За прохождение %d уровня можно получить %d очков\n", level, reward)

	attempts := 0
	for {
		attempts++
		fmt.Printf("Твоя %d попытка - ", attempts)
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		guess, err := strconv.Atoi(input)
		if err != nil {
			fmt.Println("Ошибка! Введите целое число")
			attempts--
			continue
		}

		if guess > secret {
			fmt.Println("Много. Попробуй ввести число меньшего значения")
		} else if guess < secret {
			fmt.Println("Мало. Попробуй ввести число большего значения")
		} else {
			points := reward - ((attempts - maxAttempts) * 100)
			if points < 0 {
				points = 0
			}
			balance += points
			fmt.Println("\n______________________________________________________\n")
			fmt.Printf("Молодец! Ты угадал число. Твой счёт увеличился на %d очков\n", points)
			break
		}
	}
}

File: Game.py
print("При обнаружении багов пишите мне на почту - pythonerkk@mail.ru")
print("Нужно кинуть скрин недоработки")
print()
print("______________________________________________________")
print()

from random import randint
balans = 0
# Вход + добавить цикл ко всей игре и условие выхода из игры
def Game():
    global balans
    print('Выбери уровень сложности:')
    print('1 уровень от 1 до 100')
    print('2 уровень от 1 до 250')
    print('3 уровень от 1 до 500')
    print("4 уровень от 1 до 1000")
    print("5 уровень от 1 до 2000")
    g = False
    while g != True:
        try:
            y = int(input("Введи номер уровня: "))
            if y == 1 or y == 2 or y ==3 or y == 4 or y == 5:
                g =True
            else:
                print("Введите целое число от 1 до 5")
        except ValueError:
            print('Ошибка! Введите целое число от 1 до 5')


    # 1 уровень
    if y == 1:
        x = randint(1, 100)
        g = False
        print('Угадай число от 1 до 100 за 6 попыток. За лишнию попытку у тебя будет сниматься по 100 очков')
        print("За прохождение 1 уровня можно получить 500 очков")
        i = 0
        while g != True:
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                g =True
            except ValueError:
                print('Ошибка! Введите целое число')
        h = True
        while a!=x:
            if h == True:
                if a > x:
                    print('Много. Попробуй ввести число меньшего значения')
                elif a < x:
                    print('Мало. Попробуй ввести число большего значения')
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                h = True
            except ValueError:
                print('Ошибка! Введите целое число')
                h = False
        if a == x:
            if i <= 6:
                count = 500
            else:
                count = 500 - ((i - 6) * 100)
                if count < 0:
                    count = 0
            balans += count
            print()
            print("______________________________________________________")
            print()
            print('Молодец! Ты угадал число. Твой счёт увеличился на',count, "очков")


    # 2 уровень
    if y == 2:
        x = randint(1, 250)
        g = False
        print('Угадай число от 1 до 250 за 7 попыток. За лишнию попытку у тебя будет сниматься по 100 очков')
        print("За прохождение 2 уровня можно получить 800 очков")
        i = 0
        while g != True:
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                g =True
            except ValueError:
                print('Ошибка! Введите целое число')
        h = True
        while a!=x:
            if h == True:
                if a > x:
                    print('Много. Попробуй ввести число меньшего значения')
                elif a < x:
                    print('Мало. Попробуй ввести число большего значения')
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                h = True
            except ValueError:
                print('Ошибка! Введите целое число')
                h = False
        if a == x:
            if i <= 7:
                count = 800
            else:
                count = 800 - ((i - 7) * 100)
                if count < 0:
                    count = 0
            balans += count
            print()
            print("______________________________________________________")
            print()
            print('Молодец! Ты угадал число. Твой счёт увеличился на',count, "очков")

    # 3 уровень
    if y == 3:
        x = randint(1, 500)
        g = False
        print('Угадай число от 1 до 500 за 8 попыток. За лишнию попытку у тебя будет сниматься по 100 очков')
        print("За прохождение 3 уровня можно получить 1000 очков")
        i = 0
        while g != True:
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                g = True
            except ValueError:
                print('Ошибка! Введите целое число')
        h = True
        while a != x:
            if h == True:
                if a > x:
                    print('Много. Попробуй ввести число меньшего значения')
                elif a < x:
                    print('Мало. Попробуй ввести число большего значения')
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                h = True
            except ValueError:
                print('Ошибка! Введите целое число')
                h = False
        if a == x:
            if i <= 8:
                count = 1000
            else:
                count = 1000 - ((i - 8) * 100)
                if count < 0:
                    count = 0
            balans += count
            print()
            print("______________________________________________________")
            print()
            print('Молодец! Ты угадал число. Твой счёт увеличился на',count, "очков")


    #4 уровень
    if y == 4:
        x = randint(1, 1000)
        g = False
        print('Угадай число от 1 до 1000 за 9 попыток. За лишнию попытку у тебя будет сниматься по 100 очков')
        print("За прохождение 4 уровня можно получить 1200 очков")
        i = 0
        while g != True:
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                g =True
            except ValueError:
                print('Ошибка! Введите целое число')
        h = True
        while a!=x:
            if h == True:
                if a > x:
                    print('Много. Попробуй ввести число меньшего значения')
                elif a < x:
                    print('Мало. Попробуй ввести число большего значения')
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                h = True
            except ValueError:
                print('Ошибка! Введите целое число')
                h = False
        if a == x:
            if i <= 9:
                count = 1200
            else:
                count = 1200 - ((i - 9) * 100)
                if count < 0:
                    count = 0
            balans += count
            print('Молодец! Ты угадал число. Твой счёт увеличился на',count, "очков")

    #5 уровень
    if y == 5:
        x = randint(1, 2000)
        g = False
        print('Угадай число от 1 до 2000 за 10 попыток. За лишнию попытку у тебя будет сниматься по 100 очков')
        print("За прохождение 5 уровня можно получить 1500 очков")
        i = 0
        while g != True:
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                g =True
            except ValueError:
                print('Ошибка! Введите целое число')
        h = True
        while a!=x:
            if h == True:
                if a > x:
                    print('Много. Попробуй ввести число меньшего значения')
                elif a < x:
                    print('Мало. Попробуй ввести число большего значения')
            i += 1
            try:
                a = int(input(f'Твоя {i} попытка - '))
                h = True
            except ValueError:
                print('Ошибка! Введите целое число')
                h = False
        if a == x:
            if i <= 10:
                count = 1500
            else:
                count = 1500 - ((i - 10) * 100)
                if count < 0:
                    count = 0
            balans += count
            print('Молодец! Ты угадал число. Твой счёт увеличился на',count, "очков")

print("Твой общий счёт =", balans)
print()
print("______________________________________________________")
print()
Game()
close = ""
while close.lower() != "нет":
    print()
    print("______________________________________________________")
    print()
    print("Твой общий счёт =", balans)
    print("Продолжаем игру?")
    close = input()
    print()
    print("______________________________________________________")
    print()
    if close.lower() == "да":
        Game()
    elif close.lower() == "нет":
        continue
    else:
        print("Напиши да или нет")
print("Для закрытия консоли нажми 'ENTER'")
print("Made by MicheZZ")
input()




